src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -263,16 +363,32 @@ public String getDisplayName() {
      * @throws FormException if the formData is invalid.
      */
     @Override
+    @SuppressFBWarnings("REC_CATCH_EXCEPTION")
     public boolean configure(final StaplerRequest req, final JSONObject formData) throws FormException {
         try {
-            super.configure(req, formData);
+            Boolean status = super.configure(req, formData);
 
-            // Grab apiKey and hostname
             this.setReportWith(formData.getString("reportWith"));
             this.setTargetApiURL(formData.getString("targetApiURL"));
+            this.setTargetLogIntakeURL(formData.getString("targetLogIntakeURL"));
             this.setTargetApiKey(formData.getString("targetApiKey"));
             this.setTargetHost(formData.getString("targetHost"));
-            this.setTargetPort(formData.getInt("targetPort"));
+            String portStr = formData.getString("targetPort");
+            if(StringUtils.isNotBlank(portStr)){
+                if(StringUtils.isNumeric(portStr)) {
+                    this.setTargetPort(formData.getInt("targetPort"));
+                }
+            }else{
+                this.setTargetPort(null);
+            }
+            String logCollectionPortStr = formData.getString("targetLogCollectionPort");
+            if(StringUtils.isNotBlank(logCollectionPortStr)){
+                if(StringUtils.isNumeric(logCollectionPortStr)){
+                    this.setTargetLogCollectionPort(formData.getInt("targetLogCollectionPort"));
+                }
+            }else{
+                this.setTargetLogCollectionPort(null);
+            }
```

Comment:
isn't it supposed to be validated by `validateTargetPort` ?

Reply:
yes and it is happening below.

Reply:
I guess I don’t understand why we do the type checking here as well